app/assets/javascripts/class_lists/ExportList.js

```diff
@@ -0,0 +1,254 @@
+import React from 'react';
+import _ from 'lodash';
+import DownloadCsvLink from '../components/DownloadCsvLink';
+import Button from '../components/Button';
+import SuccessLabel from '../components/SuccessLabel';
+import tableStyles from '../components/tableStyles';
+import {gradeText} from '../helpers/gradeText';
+import {
+  UNPLACED_ROOM_KEY,
+  createRooms,
+  findMovedStudentIds
+} from './studentIdsByRoomFunctions';
+
+
+// Allows principals to set the names of teachers for each room, and export
+// to a spreadsheet.  Others can view but can't export if the principal hasn't
+// added names yet.
+// The CSV download is within the browser.
+export default class ExportList extends React.Component {
+  mapToRows(studentIdsByRoom) {
+    const {gradeLevelNextYear, students, principalTeacherNamesByRoom} = this.props;
+
+    return _.flatten(_.compact(Object.keys(studentIdsByRoom).map(roomKey => {
+      return studentIdsByRoom[roomKey].map(studentId => {
+        const student = _.find(students, {id: studentId});
+        const teacherText = principalTeacherNamesByRoom[roomKey] || 'Not placed';
+        return [
+          gradeLevelNextYear,
+          student.local_id,
+          `${student.first_name} ${student.last_name}`,
+          teacherText
+        ];
+      });
+    })));
+  }
+
+  isReadyToExport(studentIdsByRoom) {
+    const {principalTeacherNamesByRoom} = this.props;
+    const isMissingTeacherNames = (Object.keys(principalTeacherNamesByRoom).length < Object.keys(studentIdsByRoom).length - 1);
+    const hasBlankTeacherName = _.any(_.values(principalTeacherNamesByRoom), _.isEmpty);
+
+    return (!isMissingTeacherNames && !hasBlankTeacherName);
+  }
+
+  onRoomTeacherChanged(roomKey, e) {
+    e.preventDefault();
+
+    const {principalTeacherNamesByRoom, onPrincipalTeacherNamesByRoomChanged} = this.props;
+    onPrincipalTeacherNamesByRoomChanged({
+      ...principalTeacherNamesByRoom,
+      [roomKey]: e.target.value
+    });
+  }
+
+  // Button onClick does nothing, since DownloadCsvLink handles it,
+  // we're just using Button for the visual.
+  onDownloadButtonClicked(e) {}
+
+  render() {
+    const {
+      headingStyle,
+      teacherStudentIdsByRoom,
+      principalStudentIdsByRoom
+    } = this.props;
+    const studentIdsByRoom = principalStudentIdsByRoom || teacherStudentIdsByRoom; // TODO(kr)
+    const rows = this.mapToRows(studentIdsByRoom);
+
+    return (
+      <div className="SecretaryEnters">
+        <div style={headingStyle}>Have all students been placed?</div>
+        <div style={styles.spaceBelow}>{this.renderUnplaced(studentIdsByRoom)}</div>
+        
+        <div style={headingStyle}>Have you reviewed the teaching team's plan and notes before making changes?</div>
```

Comment:
Yeah this is a pain.  I googled for minute yesterday and couldn't figure it out but we should definitely sort this out since it makes reviewing these diffs super tough...